File: 剑指offer/50. 第一个只出现一次的字符.py
########哈希表的使用

class Solution:
    def firstUniqChar(self, s: str) -> str:
        dic = {}
        for c in s:
            dic[c] = dic.get(c,0) + 1
        for c in s:
            if dic[c] == 1: return c
        return " "
        
     https://leetcode-cn.com/problems/di-yi-ge-zhi-chu-xian-yi-ci-de-zi-fu-lcof/solution/mian-shi-ti-50-di-yi-ge-zhi-chu-xian-yi-ci-de-zi-3/
        
        

File: 剑指offer/C++_Version/50. 第一个只出现一次的字符.cpp
//方法1

class Solution {
public:
    char firstUniqChar(string s) 
    {
        unordered_map<char, int> dic;
        for(int i=0; i<s.size(); i++)
            ++dic[s[i]]; // 写成++dic[s[i]]也是对的，但是要好好理解下i++, 与++i的区别;
        for(char c : s)
            if(dic[c]==1)
                return c;
        return ' ';
    }
    
};
